test/e2e/nodepool_test.go

```diff
@@ -44,9 +46,10 @@ func TestNodePool(t *testing.T) {
 	guestCluster := e2eutil.CreateCluster(t, ctx, mgmtClient, &clusterOpts, globalOpts.Platform, globalOpts.ArtifactDir)
 	guestClient := e2eutil.WaitForGuestClient(t, ctx, mgmtClient, guestCluster)
 
-	// Set of tests
-	// Each test should have their own NodePool
-	t.Run("TestNodePoolAutoRepair", testNodePoolAutoRepair(ctx, mgmtClient, guestCluster, guestClient, clusterOpts))
+	t.Run("Refactored", func(t *testing.T) {
```

Comment:
Why are you creating a common context here and passing that through?
Why is that better than letting each function (testNodePoolAutoRepair, testNodepoolMachineconfigGetsRolledout...) to create their context from testContext i.e. `ctx, cancel := context.WithCancel(testContext)`?

Reply:
AFAIK I'm creating a context in the test group function in order to isolate from the general one, and if the context get cancelled, it should only affect their group set. Also for convention, we are following that procedure in the E2E chaos_test.

We can change that and only open a new context for every inner test and not in the test set function, inheriting from the `testContext`